platform/commonUI/edit/src/actions/SaveInProgressDialog.js

```diff
@@ -1,21 +1,30 @@
 define([], function () {
-    function SaveInProgressDialog(dialogService) {
-        this.dialogService = dialogService;
-        this.dialog = undefined;
+    var self = this;
+
+    function SaveInProgressDialog(dialogService, $timeout) {
+        self.dialogService = dialogService;
+        self.$timeout = $timeout;
+
+        self.dialog = undefined;
+        self.timeoutId = undefined;
     }
 
     SaveInProgressDialog.prototype.show = function () {
-        this.dialog = this.dialogService.showBlockingMessage({
-            title: "Saving...",
-            hint: "Do not navigate away from this page or close this browser tab while this message is displayed.",
-            unknownProgress: true,
-            severity: "info"
-        });
+        self.$timeout.cancel(self.timeoutId);
```

Comment:
The `self` variable could be scoped to this function, and then `this` will be an instance of `SaveInProgressDialog` object.